src/plugins/intel_cpu/src/nodes/matrix_nms.cpp

```diff
@@ -370,7 +374,8 @@ void MatrixNms::execute(dnnl::stream strm) {
     auto selectedIndicesMemPtr = getChildEdgesAtPort(NMS_SELECTED_INDICES)[0]->getMemoryPtr();
     auto validOutputsMemPtr = getChildEdgesAtPort(NMS_VALID_OUTPUTS)[0]->getMemoryPtr();
 
-    // TODO [DS NMS]: remove when nodes from models where nms is not last node in model supports DS
+    // NMS-alike nodes are always transformed to NMSIEInternal node in case of legacy api, for compatibility.
+    // And on the other hand in case of api 2.0, keep them internal dynamic for better performance and functionality.
     if (isDynamicNode()) {
```

Comment:
Shouldn't there be condition that depends on m_outStaticShape field?

Reply:
yes, you are right. thanks for pointing this.